challenge-7.js

```diff
@@ -0,0 +1,193 @@
+const warehouse = Warehouse([]);
+
+function Product(id, name, price, count, sell = [], enable = true) {
+  if (count === 0) enable = false;
+  return {
+    id,
+    name,
+    count,
+    price,
+    enable,
+    sell,
+  };
+}
+
+function Warehouse(products = []) {
+  return products;
+}
+
+function warehouseStock() {
+  let productsCount = 0;
+  for (const product of warehouse) {
+    productsCount += product.count;
+  }
+  return productsCount;
+}
+
+function productStock(productId) {
+  if (!isTypeValid(productId, "number")) return "id should be a number!";
+  for (const product of warehouse) {
+    if (product.id === productId) {
+      return product;
+    }
+  }
+  return `product ${productId} is not available!`;
+}
+
+function addNewProduct(product) {
+  if (findProduct(product.id)) return "id should be unique!";
+  warehouse.push(product);
+  return product.name + " is added!";
+}
+
+function sellProduct(productId, count = 1) {
+  if (!isTypeValid(productId, "number")) return "id should be a number!";
+  if (!isTypeValid(count, "number")) return "count must be number!";
+  if (!isInRange(count, 0)) return "count is out of range!";
+  for (const product of warehouse) {
+    if (product.id === productId && product.enable && product.count >= count) {
+      product.count -= count;
+      product.sell.push({ count, price: product.price });
+      if (product.count === 0) product.enable = false;
+      return `${count} ${product.name} sold!`;
+    }
+  }
+  return `product ${productId} is not available or enough!`;
+}
+
+function availableProduct() {
+  return warehouse
```

Comment:
`map` and `filter` does return an array and hence there is no need to use `return` keyword here.
Learn more about them [here](https://javascript.plainenglish.io/take-a-good-look-at-filter-map-and-reduce-in-javascript-18d72f483ada)